source/extensions/load_balancing_policies/client_side_weighted_round_robin/client_side_weighted_round_robin_lb.cc

```diff
@@ -0,0 +1,333 @@
+    return absl::InvalidArgumentError("Utilization must be positive");
+  }
+
+  // Calculate the weight.
+  double weight = qps / utilization;
+
+  // Limit the weight to uint32_t max.
+  if (weight > std::numeric_limits<uint32_t>::max()) {
+    weight = std::numeric_limits<uint32_t>::max();
+  }
+  return weight;
+}
+
+absl::Status ClientSideWeightedRoundRobinLoadBalancer::OrcaLoadReportHandler::
+    updateClientSideDataFromOrcaLoadReport(const OrcaLoadReportProto& orca_load_report,
+                                           ClientSideHostLbPolicyData& client_side_data) {
+  const absl::StatusOr<uint32_t> weight = calculateWeightFromOrcaReport(
+      orca_load_report, metric_names_for_computing_utilization_, error_utilization_penalty_);
+  if (!weight.ok()) {
+    return weight.status();
+  }
+
+  const MonotonicTime now = time_source_.monotonicTime();
+  // Update client side data attached to the host.
+  absl::MutexLock lock(&client_side_data.mu_);
+  client_side_data.weight_ = weight.value();
+  client_side_data.last_update_time_ = now;
+  if (client_side_data.non_empty_since_ == ClientSideHostLbPolicyData::kDefaultNonEmptySince) {
+    client_side_data.non_empty_since_ = now;
+  }
+  return absl::OkStatus();
+}
+
+Upstream::LoadBalancerPtr ClientSideWeightedRoundRobinLoadBalancer::WorkerLocalLbFactory::create(
+    Upstream::LoadBalancerParams params) {
+  const auto* typed_lb_config =
+      dynamic_cast<const ClientSideWeightedRoundRobinLbConfig*>(lb_config_.ptr());
+  return std::make_unique<Upstream::ClientSideWeightedRoundRobinLoadBalancer::WorkerLocalLb>(
+      params.priority_set, params.local_priority_set, cluster_info_.lbStats(), runtime_, random_,
+      cluster_info_.lbConfig(), typed_lb_config->lb_config_, time_source_);
+}
+
+ClientSideWeightedRoundRobinLoadBalancer::ClientSideWeightedRoundRobinLoadBalancer(
+    OptRef<const Upstream::LoadBalancerConfig> lb_config, const Upstream::ClusterInfo& cluster_info,
+    const Upstream::PrioritySet& priority_set, Runtime::Loader& runtime,
+    Envoy::Random::RandomGenerator& random, TimeSource& time_source)
+    : factory_(std::make_shared<WorkerLocalLbFactory>(lb_config, cluster_info, priority_set,
+                                                      runtime, random, time_source)),
+      lb_config_(lb_config), cluster_info_(cluster_info), priority_set_(priority_set),
+      runtime_(runtime), random_(random), time_source_(time_source) {}
+
+absl::Status ClientSideWeightedRoundRobinLoadBalancer::initialize() {
+  // Ensure that all hosts have client side lb policy data.
+  for (const HostSetPtr& host_set : priority_set_.hostSetsPerPriority()) {
+    addClientSideLbPolicyDataToHosts(host_set->hosts());
+  }
+
+  const auto* typed_lb_config =
+      dynamic_cast<const ClientSideWeightedRoundRobinLbConfig*>(lb_config_.ptr());
+  initFromConfig(typed_lb_config->lb_config_);
+  startWeightUpdatesOnMainThread(typed_lb_config->main_thread_dispatcher_);
+  return absl::OkStatus();
+}
```

Comment:
I think you forget to handle the hosts that added by the EDS at the runtime? You may need to call the `addPriorityUpdateCb` to handle the new hosts.